### PR TITLE
Adds some basic rate limiting

### PR DESCRIPTION
**Pull request by athompson-r7**

### CODE REVIEW

File: lib/convection/control/stack.rb [hunk 1]
```diff
@@ -201,7 +202,17 @@ def apply(&block)
 
         watch(&block) if block # Block execution on stack status
       rescue Aws::Errors::ServiceError => e
-        @errors << e
+        if e.message == 'Rate exceeded'
+          operation = context.operation_name
```
Comment (jmanero-r7) [inline lib/convection/control/stack.rb]: Where's `context` coming from?

Comment (athompson-r7) [inline lib/convection/control/stack.rb]: Ooops, I must have trimmed that when cleaning up.

File: lib/convection/control/stack.rb [hunk 2]
```diff
@@ -85,6 +85,7 @@ def initialize(name, template, options = {})
         @resources = {}
         @current_template = {}
         @last_event_seen = nil
+        @retries = 0
```
Comment (athompson-r7) [inline lib/convection/control/stack.rb]: I'm not exactly sure if I needed to reset `@retries` within each method that was going to use this or if we get a new instance of this class each time.

Comment (jmanero-r7) [inline lib/convection/control/stack.rb]: As far as I remember here, the Stack is only instantiated at startup when the Cloudfile is parsed.

**Comment by athompson-r7** [comment]
@jmanero-r7 simplified this a bit.

**Comment by athompson-r7** [comment]
Fixes #31

**Pull request merged**